src/emu/tilemap.h

```diff
@@ -498,10 +503,16 @@ class tilemap_t
 	running_machine &machine() const;
 	tilemap_device *device() const { return m_device; }
 	palette_device &palette() const { return *m_palette; }
+	device_gfx_interface &decoder() const { return *m_tileinfo.decoder; }
+
 	tilemap_t *next() const { return m_next; }
 	void *user_data() const { return m_user_data; }
 	memory_array &basemem() { return m_basemem; }
 	memory_array &extmem() { return m_extmem; }
+	UINT32 rows() const { return m_rows; }
```

Comment:
This whole class is kinda weird when it comes to how integers are (ab)used - the constructor takes signed integers and converts them to `UINT32` without actually doing anything special for negative values.  If you're cleaning up/changing the interface, why not clean up the misguided abuse of integers in general?